sklearn/tests/test_discriminant_analysis.py

```diff
@@ -667,3 +669,52 @@ def test_get_feature_names_out():
         dtype=object,
     )
     assert_array_equal(names_out, expected_names_out)
+
+
+@pytest.mark.parametrize("X, y", [(X, y), (X, y3)])
+def test_lda_array_api(X, y):
+    """Check that the array_api Array gives the same results as ndarrays."""
+    pytest.importorskip("numpy", minversion="1.22", reason="Requires Array API")
+    xp = pytest.importorskip("numpy.array_api")
```

Comment:
we kinda still have numpy array api specific code. I think we need something here which is not numpy's array api implementation to test.